metadata/src/main/java/org/apache/kafka/image/MetadataDelta.java

```diff
@@ -51,7 +51,7 @@
 public final class MetadataDelta {
     private final MetadataImage image;
 
-    private long highestOffset;
+    private long highestOffsetExclusive;
```

Comment:
How about `highestEndOffset`? I am suggesting the `EndOffset` suffix because I am planning of adding a type called `SnapshotId` which at high-level it will have two fields: `endOffset()` and `epoch()`.